mmengine/analysis/jit_analysis.py

```diff
@@ -0,0 +1,687 @@
+                    self._enable_warn_uncalled_mods).tracer_warnings(
+                        self._warn_trace))
+
+    def tracer_warnings(self: T, mode: str) -> T:
+        """Sets which warnings to print when tracing the graph to calculate
+        statistics. There are three modes. Defaults to 'no_tracer_warning'.
+        Allowed values are:
+
+        * 'all' : keeps all warnings raised while tracing
+        * 'no_tracer_warning' : suppress torch.jit.TracerWarning only
+        * 'none' : suppress all warnings raised while tracing
+
+        Args:
+            mode (str) : warning mode in one of the above values.
+        """
+        if mode not in ['all', 'no_tracer_warning', 'none']:
+            raise ValueError(f'Unrecognized tracer warning mode {mode}.')
+        self._warn_trace = mode
+        return self
+
+    def ancestor_mode(self: T, mode: str) -> T:
+        """Sets how to determine the ancestor modules of an operator. Must be
+        one of "owner" or "caller".
+
+        * "caller": an operator belongs to all modules that is currently
+            executing `forward()` at the time the operator is called.
+        * "owner": an operator belongs to the last module that's executing
+            `forward()` at the time the operator is called, plus this
+            module's recursive parents. If an module has multiple parents
+            (e.g. a shared module), only one will be picked.
+        For most cases, a module only calls submodules it owns, so both
+        options would work identically. In certain edge cases, this option
+        will affect the hierarchy of results, but won't affect the total
+        count.
+        """
+        if mode not in ['owner', 'caller']:
+            raise ValueError(f'Unrecognized ancestor mode: {mode}')
+        self._ancestor_mode = mode
+        return self
+
+    def unsupported_ops_warnings(self: T, enabled: bool) -> T:
+        """Sets if warnings for unsupported operators are shown.
+
+        Defaults
+        to True. Counts of unsupported operators may be obtained from
+        :meth:`unsupported_ops` regardless of this setting.
+
+        Args:
+            enabled (bool) : Set to 'True' to show unsupported operator
+                warnings.
+        """
+        self._enable_warn_unsupported_ops = enabled
+        return self
+
+    def uncalled_modules_warnings(self: T, enabled: bool) -> T:
+        """Sets if warnings from uncalled submodules are shown.
+
+        Defaults to true.
+        A submodule is considered "uncalled" if it is never called during
+        tracing. This may be because it is actually unused, or because it is
+        accessed via calls to ``.forward()`` or other methods of the module.
+        The set of uncalled modules may be obtained from
+        :meth:`uncalled_modules` regardless of this setting.
+
+        Args:
+            enabled (bool) : Set to 'True' to show warnings.
```

Comment:
```suggestion
        Defaults to true. A submodule is considered "uncalled" if it is never
        called during tracing. This may be because it is actually unused, or
        because it is accessed via calls to ``.forward()`` or other methods of
        the module. The set of uncalled modules may be obtained from
        :meth:`uncalled_modules` regardless of this setting.

        Args:
            enabled (bool): Set to 'True' to show warnings.
```